salt/metalk8s/volumes/prepared/files/metalk8s-sparse-volume-cleanup

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python2
+
+from __future__ import print_function
+import argparse
+import errno
+import os
+import stat
+import sys
+
+
+SPARSE_FILES_DIR = '/var/lib/metalk8s/storage/sparse/'
+LOOP_MAJOR = 7
+
+
+def parse_args(args=None):
+    parser = argparse.ArgumentParser(
+        prog='metalk8s-sparse-volume-cleanup',
+        help='Cleanup MetalK8s sparse loop volumes',
+    )
+
+    parser.add_argument(
+        '-i', '--uid',
+        required=True,
+        help='UID of the MetalK8s Volume object',
+        dest='volume_id',
+    )
+
+    return parser.parse_args(args=args)
+
+
+def check_sparse_file(volume_id):
+    """Check if the sparse file exists for this volume ID."""
+    sparse_file_path = SPARSE_FILES_DIR + volume_id
+    if not os.path.isfile(sparse_file_path):
+        die(
+            "Sparse file {} does not exist for volume '{}'".format(
+                sparse_file_path, volume_id
+            ),
+            exit_code=errno.ENOENT
+        )
+
+
+def cleanup(volume_id):
+    device_path = find_device(volume_id)
+
+    print("Detaching loop device '{}' for volume '{}'".format(
+        device_path, volume_id
+    ))
+
+    fd = os.open(device_path, os.O_RDONLY)
+    try:
+        fcntl.ioctl(fd, LOOP_CLR_FD, 0)
+    except IOError as exn:
+        if exn.errno != errno.ENXIO:
+            die("Unexpected error when trying to free device {}: {}".format(
+                device_path, str(exn)
+            ))
+        print("Device already freed", file=sys.stderr)
+    finally:
+        fd.close()
+
+    print("Loop device for volume '{}' was successfully detached".format(
+        volume_id
+    ))
+
+
+def main():
+    args = parse_args()
+
+    check_sparse_file(args.volume_id)
+    cleanup(args.volume_id)
+
+
+# Helpers {{{
+def die(message, exit_code=1):
+    print(message, file=sys.stderr)
```

Comment:
Does `sys.stderr` need a `flush` before `sys.exit`?

Reply:
`stderr` is usually unbuffered, but [it's not that clear in Python](https://stackoverflow.com/a/22478875) so we may want to flush to be safe.